### ISSUE TITLE
postcss-sorting complains about `less-mixins`

### ISSUE THREAD

**Issue by stephanschubert** [hudochenkov/stylelint-order#71]
When `less-mixins` is used:

```
    "order/order": [
      "custom-properties",
      "dollar-variables",
      "at-variables",
      "declarations",
      "rules",
      "at-rules",
      "less-mixins",
    ],
    "order/properties-alphabetical-order": true,
```

Result:

```
postcss-sorting: order: Every item in the array must be an object with a "type" property, or one of keywords: custom-properties, dollar-variables, at-variables, declarations, rules, at-rules.
```

Removing `less-mixins` "fixes" the error.

**Comment by hudochenkov** [comment]
Thank you for reporting!

Indeed, this problem exists, because `postcss-sorting` (a plugin which handles order fixing) doesn't support this keyword.

I believe it could be fixed if add to https://github.com/hudochenkov/postcss-sorting/blob/944da947a628192b54448368c197f586fbbe0c10/lib/getOrderData.js code similar to https://github.com/hudochenkov/stylelint-order/blob/3215ac3dee90f7ebfb48fd36601727275a613ca2/rules/order/getOrderData.js#L19-L20

Consider contributing this fix, because Less is not in my priority.

While problem exists you can [disable autofixing](https://github.com/hudochenkov/stylelint-order#autofixing) for `order/order`.

**Comment by NE-SmallTown** [comment]
Same problem.

@hudochenkov  The bad I feel is that this will cause the entire file won't be fixed, I think we should just ignore `less-mixins` and still fix the file and **throw the error rather than just log**(for me, I even don't see the warning and spend 2 hours to dig into the source code of `stylelint-order`)

What's your suggestion/plan to fix this? If @stephanschubert doesn't have time, I could send a PR for this.

**Comment by hudochenkov** [comment]
@NE-SmallTown Sorry, you faced this situation. It's a rare case when something was added to stylelint-order but wasn't reflected in postcss-sorting. My suggestion is to add support for `less-mixins` into postcss-sorting. I [wrote](https://github.com/hudochenkov/stylelint-order/issues/71#issuecomment-432217119) above where to start.

**Comment by NE-SmallTown** [comment]
Never mind, I will take a look if @stephanschubert doesn't have time

**Comment by oller** [comment]
The ol' 18month later bump.  I'm encountering this issue too, is it as easy a fix as the snippet in the thread above?

**Comment by hudochenkov** [comment]
Pretty much. Add new keyword and add tests for it.

**Comment by oller** [comment]
Good stuff, shall try and find some time tomorrow.

**Comment by Astrael1** [comment]
Fixing this requires adding some LESS parser to [postcss-sorting](https://github.com/hudochenkov/postcss-sorting) e.g. [postcss-less](https://github.com/shellscape/postcss-less) because postcss parser doesn't accept usage of less mixins and throws error
```
.le_mixin {
  color: red;
}

a {
  .le_mixin();
}
```
@hudochenkov do you allow adding that dependency or shall we just close this issue?

**Comment by hudochenkov** [comment]
I don't think it is needed. For 6 years this issue is open it only got 3 reactions.

But I'm open for a PR to `postcss-sorting` as described above.
